content/en/blog/_posts/2021-11-18-persistentvolume-reclaim-improvements.md

```diff
@@ -0,0 +1,187 @@
+
+## Volume reclaim with Kubernetes v1.23
+
+The new behavior ensures that the underlying storage object is deleted from the backend when users attempt to delete a PV manually.
+
+**The fix is introduced as an alpha in 1.23 under the feature-gate `HonorPVReclaimPolicy`
+which is turned `off` by default**. You would also need to **update the `external-provisioner` 
+to the `v4.0.0` and enable the feature gate.**
+
+#### How does it work?
+
+The new behavior is achieved by adding a finalizer `external-provisioner.volume.kubernetes.io/finalizer` on new and existing PVs, the finalizer is only removed after the storage from backend is deleted.
+
+An example of a PV with the finalizer, notice the new finalizer in the finalizers list
+
+```
+kubectl get pv pvc-a7b7e3ba-f837-45ba-b243-dec7d8aaed53 -o yaml
+```
+
+```yaml
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  annotations:
+    pv.kubernetes.io/provisioned-by: csi.vsphere.vmware.com
+  creationTimestamp: "2021-11-17T19:28:56Z"
+  finalizers:
+  - kubernetes.io/pv-protection
+  - external-provisioner.volume.kubernetes.io/finalizer
+  name: pvc-a7b7e3ba-f837-45ba-b243-dec7d8aaed53
+  resourceVersion: "194711"
+  uid: 087f14f2-4157-4e95-8a70-8294b039d30e
+spec:
+  accessModes:
+  - ReadWriteOnce
+  capacity:
+    storage: 1Gi
+  claimRef:
+    apiVersion: v1
+    kind: PersistentVolumeClaim
+    name: example-vanilla-block-pvc
+    namespace: default
+    resourceVersion: "194677"
+    uid: a7b7e3ba-f837-45ba-b243-dec7d8aaed53
+  csi:
+    driver: csi.vsphere.vmware.com
+    fsType: ext4
+    volumeAttributes:
+      storage.kubernetes.io/csiProvisionerIdentity: 1637110610497-8081-csi.vsphere.vmware.com
+      type: vSphere CNS Block Volume
+    volumeHandle: 2dacf297-803f-4ccc-afc7-3d3c3f02051e
+  persistentVolumeReclaimPolicy: Delete
+  storageClassName: example-vanilla-block-sc
+  volumeMode: Filesystem
+status:
+  phase: Bound
+```
+
+The presence of the finalizer prevents the PV object from being removed from the cluster. As stated previously, the finalizer is only removed from the PV object after it is successfully deleted from the storage backend. To learn more about finalizers please refer [Using Finalizers to Control Deletion](https://kubernetes.io/blog/2021/05/14/using-finalizers-to-control-deletion/).
+
+#### What about migrated volumes?
+
+The fix is applicable to migrated volumes as well. However, when the feature `HonorPVReclaimPolicy` is enabled on 1.23, and migration is disabled, the finalizer is removed from the PV object if it exists.
+
+### Some caveats
+
+1. The fix is applicable only to CSI volumes and migrated volumes. In-tree volumes will exhibit older behavior.
+2. The finalizer will not be proactively added on all existing PVs, it will only be added when the `external-provisioner` syncs the PV. The `external-provisioner` syncs the PV if there are any operation that directly or indirectly triggers an PV object update.
```

Comment:
I am not sure if this is a caveat. External-provisioner will sync all PVs when it starts (i.e. when the feature gate is enabled).

Instead, I would mention that alpha feature gate must be enabled in the external-provisioner sidecar of a CSI driver to enable this feature. Bu default, the feature is off there.